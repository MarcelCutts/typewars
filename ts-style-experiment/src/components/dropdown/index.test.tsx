import * as Adapter from "enzyme-adapter-react-16";
import * as React from "react";
import Dropdown from ".";

import { configure, mount, shallow } from "enzyme";

configure({ adapter: new Adapter() });

describe("Dropdown component", () => {
  test("renders", () => {
    const component = shallow(<Dropdown />);
    expect(component).toMatchSnapshot();
  });

  test("calls handler onBlur", () => {
    const mockCallback = jest.fn().mockImplementation(e => undefined);
    const component = mount(<Dropdown callback={mockCallback} />);
    component.simulate("change", { target: { value: "option2" } });
    expect(mockCallback).toHaveBeenCalled();
  });
});
