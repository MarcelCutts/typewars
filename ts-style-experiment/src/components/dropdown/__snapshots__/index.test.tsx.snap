// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown component renders 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Dropdown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <option
          value="option1"
        >
          option1
        </option>,
        <option
          value="option2"
        >
          option2
        </option>,
        <option
          value="option3"
        >
          option3
        </option>,
      ],
      "onChange": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "option1",
          "value": "option1",
        },
        "ref": null,
        "rendered": "option1",
        "type": "option",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "option2",
          "value": "option2",
        },
        "ref": null,
        "rendered": "option2",
        "type": "option",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "option3",
          "value": "option3",
        },
        "ref": null,
        "rendered": "option3",
        "type": "option",
      },
    ],
    "type": "select",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <option
            value="option1"
          >
            option1
          </option>,
          <option
            value="option2"
          >
            option2
          </option>,
          <option
            value="option3"
          >
            option3
          </option>,
        ],
        "onChange": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "option1",
            "value": "option1",
          },
          "ref": null,
          "rendered": "option1",
          "type": "option",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "option2",
            "value": "option2",
          },
          "ref": null,
          "rendered": "option2",
          "type": "option",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "option3",
            "value": "option3",
          },
          "ref": null,
          "rendered": "option3",
          "type": "option",
        },
      ],
      "type": "select",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
